pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +272,98 @@ func Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(in *v1.PodTemplateSpec,
 	return nil
 }
 
+// helper for v1.PodIPInfo list to core.PodIPinfoList
+func v1_PodIPs_To_core_PodIPs(inList []v1.PodIPInfo) []core.PodIPInfo {
+	out := make([]core.PodIPInfo, len(inList))
+	for idx, value := range inList {
+		out[idx] = core.PodIPInfo{
+			IP:         value.IP,
+			Properties: value.Properties,
+		}
+	}
+	return out
+}
+
+// Converts podStatus from v1 to core
+// auto + handling podIp/podIps
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// handle ip/ips fields
+	if 0 < len(in.PodIP) && 0 < len(in.PodIPs) {
+		// both fields are provided
+		if in.PodIP != in.PodIPs[0].IP {
+			return fmt.Errorf("conversion Error: in.PodIP(%v) != in.PodIPs[0](%v)", in.PodIP, in.PodIPs[0].IP)
+		}
+		out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+	} else {
+		if 0 < len(in.PodIP) && 0 == len(in.PodIPs) {
+			// PodIP + NO entries in PodIPs
+			out.PodIPs = []core.PodIPInfo{
+				{
+					IP:         in.PodIP,
+					Properties: make(map[string]string),
+				},
+			}
+		} else {
+			// in.PodIP has no value
+			// copy the in.PodIPs
+			out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
+		}
+	}
+
+	// Verify no duplicates in core.PodIPs
+	// We compare IPs not the entire struct
+	ips := make(map[string]core.PodIPInfo)
+	// filter out repeated
+	for _, value := range out.PodIPs {
+		if _, ok := ips[value.IP]; !ok {
+			ips[value.IP] = value
+		}
+	}
+	// create a new list of non_repeated
+	non_repeated_list := make([]core.PodIPInfo, len(ips))
+	counter := 0
+	for _, value := range ips {
+		non_repeated_list[counter] = value
+		counter++
+	}
+	// set
+	out.PodIPs = non_repeated_list
+
+	return nil
+}
+
+// helper converts list of PodIPs from core to v1
+func core_PodIPs_To_v1_PodIPs(inList []core.PodIPInfo) []v1.PodIPInfo {
+	out := make([]v1.PodIPInfo, len(inList))
+	for idx, value := range inList {
+		out[idx] = v1.PodIPInfo{
+			IP:         value.IP,
+			Properties: value.Properties,
+		}
+	}
+	return out
+}
+
+// Converts podStatus from core to v1
+// auto + handling podIp/podIps
+func Convert_core_PodStatus_To_v1_PodStatus(in *core.PodStatus, out *v1.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_core_PodStatus_To_v1_PodStatus(in, out, s); err != nil {
+		return err
+	}
+	// handle ip/ips fields
+	//v1 gets the out.PodIps[0]
+	//v1 gets the entire list of PodIPs
+	if 0 < len(in.PodIPs) {
+		out.PodIP = in.PodIPs[0].IP
+	}
+	out.PodIPs = core_PodIPs_To_v1_PodIPs(in.PodIPs)
```

Comment:
as above, this should already be part of autoConvert (and it does seem to be, via an unsafe pointer, which is suspect to me but not this PR's fault :)

Reply:
083bf4e moves this to `autoConvert_*` family of functions